src/utils.ts

```diff
@@ -191,6 +208,18 @@ export default function utils(currentSettings: Settings) {
 					continue;
 				}
 
+				// Detect pass-through indication
+				if (argument === '--') {
+					// Error if this command does not accept pass-through arguments
+					if (!mergedSpec.passThrough) {
+						throw new ErrorWithoutStack(`Unrecognized argument: ${argument}`);
```

Comment:
This error should be more helpful. Try:

```suggestion
						throw new ErrorWithoutStack('This command does not support pass-through arguments');
```